### PR TITLE
Fix/Dlink DES3526

### PR DESCRIPTION
**Pull request by louismunro**
Improved support fixes bugs with recent firmwares.

Should fix 1766: http://packetfence.org/bugs/view.php?id=1766

### CODE REVIEW

File: lib/pf/Switch/Dlink/DES_3526.pm [hunk 1]
```diff
@@ -17,15 +21,110 @@ use Log::Log4perl;
...
+
+
+        # linkup/linkdown
+        if ( !defined $ifIndex && $field =~ $linkstatus_re ) {
+            ( $ifIndex, $op ) = ( $1, $2 );
+            $trapHashRef->{'trapType'} = 'mac';
+            $trapHashRef->{'trapIfIndex'} = $ifIndex;
+
+            if ( $op == 1 ) {
+                $trapHashRef->{'trapOperation'} = 'up';
```
Comment (julsemaan) [inline lib/pf/Switch/Dlink/DES_3526.pm]: trapType has to be used. same for the other up/down

File: lib/pf/Switch/Dlink/DES_3526.pm [hunk 2]
```diff
@@ -17,15 +21,110 @@ use Log::Log4perl;
...
+                $trapHashRef->{'trapOperation'} = 'down';
+            }
+            else {
+                $trapHashRef->{'trapOperation'} = 'unknown';
+            }
+        }
+    }
+
+    unless ( $ifIndex && $op ) {
+        $trapHashRef->{'trapOperation'} = 'unknown';
```
Comment (julsemaan) [inline lib/pf/Switch/Dlink/DES_3526.pm]: trapType

File: lib/pf/Switch/Dlink/DES_3526.pm [hunk 3]
```diff
@@ -9,6 +9,10 @@ pf::Switch::Dlink::DES_3526 - Object oriented module to access SNMP enabled Dlin
 The pf::Switch::Dlink::DES_3526 module implements an object oriented interface
 to access SNMP enabled Dlink DES 3526 switches.
 
+Tested on Firmware: Build 5.01.B65.
+Supports MAC detection traps with linkup/linkdown.
+No portsecurity support, no RADIUS.
```
Comment (extrafu) [inline lib/pf/Switch/Dlink/DES_3526.pm]: portsecurity -> port security

File: lib/pf/Switch/Dlink/DES_3526.pm (hunk 1 above)
Comment (louismunro) [inline lib/pf/Switch/Dlink/DES_3526.pm]: Can you merge you changes with this pull request please?

File: lib/pf/Switch/Dlink/DES_3526.pm [hunk 4]
```diff
@@ -17,15 +21,110 @@ use Log::Log4perl;
...
+    }
+
+    return $trapHashRef;
+}
 
 =head1 AUTHOR
 
-Treker Chen <treker.chen@gmail.com>
+Treker Chen <treker.chen@gmail.com> for the original implementation.
+Louis Munro <lmunro@inverse.ca> for the updated version.
```
Comment (extrafu) [inline lib/pf/Switch/Dlink/DES_3526.pm]: We avoid putting personal names. Just add:

Inverse inc. info@inverse.ca for the updated version.

File: lib/pf/Switch/Dlink/DES_3526.pm [hunk 5]
```diff
@@ -17,15 +21,110 @@ use Log::Log4perl;
...
 
 =head1 COPYRIGHT
 
-Copyright (C) 2008 Treker Chen
+Copyright (C) Inverse inc.
```
Comment (extrafu) [inline lib/pf/Switch/Dlink/DES_3526.pm:98]: Copyright (C) 2014 Inverse inc.

**Comment by julsemaan** [comment]
I had to flush the up/down for the 3028 because the switch was getting confused. I'm wondering if they are really necessary for this switch module also

File: lib/pf/Switch/Dlink/DES_3526.pm (hunk 1 above)
Comment (louismunro) [inline lib/pf/Switch/Dlink/DES_3526.pm]: Never mind. I misunderstood your comment.

**Pull request merged**